cvmfs/util/algorithm.h

```diff
@@ -108,6 +115,59 @@ class StopWatch : SingleCopy {
 };
 
 
+class Log2Hist
+{
+ private:
+  uint number_of_bins;
+  atomic_int32 *bins;
+  uint *boundary_values;
+  uint count_digits(uint64_t n)
+  {
+    return (uint)floor(log10(n) + 1);
+  }
+  string generate_stars(uint n)
+  {
+    uint i = 0;
+    string stars = "";
+    for (i = 0; i < n; i++)
+    {
+      stars += "*";
+    }
+    return stars;
+  }
+  string to_string(uint n)
+  {
+    stringstream s;
+    s << n;
+    return s.str();
+  }
+ public:
+  explicit Log2Hist(uint n);
+  ~Log2Hist();
+  void Add(float value)
+  {
+    uint i;
+    uint flag = 1;
+
+    for (i = 1; i <= this->number_of_bins; i++)
```

Comment:
I think it would be easier to make the last bin an overflow bin and set its boundary value to the maximum value.  You then would not need a special flag.

Reply:
The thing is that the boundary values are stored in the following way: [0, 2, 4, 8, 16 ...]. The point of storing the boundary values is that later, when I print the histogram, it is easy to avoid large if statement for a special case "0 -> 1". If I change the overflow bin to be the last one, then I need to fix boundary_value array which will take some time. Due to the lack of time, I will keep it like that. Later, if we reach the goal of the project, I can modify and make pull request once again from home. I do not think that current implementation will cause major  performance issues since it will be used just for printing the metrics.